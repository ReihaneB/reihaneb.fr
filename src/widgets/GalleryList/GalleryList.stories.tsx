import type { Meta, StoryObj } from '@storybook/react';

import _16x9 from '@/styles/assets/placeholder/_16x9.png';
import _4x3 from '@/styles/assets/placeholder/_4x3.png';
import _1x1 from '@/styles/assets/placeholder/_1x1.png';

import GalleryList from './GalleryList';

const meta: Meta<typeof GalleryList> = {
  title: 'Widgets/GalleryList',
  component: GalleryList,
};

export default meta;
type Story = StoryObj<typeof GalleryList>;

const defaultArgs = {
  galleries: [
    {
      id: 1,
      name: 'Gallery 1',
      images: [
        {
          id: 1,
          src: _1x1,
          alt: 'Placeholder image',
        },
        {
          id: 2,
          src: _16x9,
          alt: 'Placeholder image',
        },
      ],
    },
    {
      id: 2,
      name: 'Gallery 2',
      images: [
        {
          id: 1,
          src: _16x9,
          alt: 'Placeholder image',
        },
        {
          id: 2,
          src: _16x9,
          alt: 'Placeholder image',
        },
        {
          id: 3,
          src: _4x3,
          alt: 'Placeholder image',
        },
      ],
    },
  ],
};

export const Playground: Story = {
  args: {
    ...defaultArgs,
  },
};

export const _1Gallery: Story = {
  args: {
    galleries: [
      {
        id: 1,
        name: 'Gallery 1',
        images: [
          {
            id: 1,
            src: _16x9,
            alt: 'Placeholder image',
          },
          {
            id: 2,
            src: _16x9,
            alt: 'Placeholder image',
          },
        ],
      },
    ],
  },
};

export const _2Galleries: Story = {
  args: {
    galleries: [
      {
        id: 1,
        name: 'Gallery 1',
        images: [
          {
            id: 1,
            src: _16x9,
            alt: 'Placeholder image',
          },
          {
            id: 2,
            src: _16x9,
            alt: 'Placeholder image',
          },
        ],
      },
      {
        id: 2,
        name: 'Gallery 2',
        description: 'Description',
        images: [
          {
            id: 1,
            src: _16x9,
            alt: 'Placeholder image',
          },
          {
            id: 2,
            src: _16x9,
            alt: 'Placeholder image',
          },
          {
            id: 3,
            src: _4x3,
            alt: 'Placeholder image',
          },
        ],
      },
    ],
  },
};

export const _3Galleries: Story = {
  args: {
    galleries: [
      {
        id: 1,
        name: 'Gallery 1',
        images: [
          {
            id: 1,
            src: _16x9,
            alt: 'Placeholder image',
          },
          {
            id: 2,
            src: _16x9,
            alt: 'Placeholder image',
          },
        ],
      },
      {
        id: 2,
        name: 'Gallery 2',
        images: [
          {
            id: 1,
            src: _16x9,
            alt: 'Placeholder image',
          },
          {
            id: 2,
            src: _16x9,
            alt: 'Placeholder image',
          },
          {
            id: 3,
            src: _4x3,
            alt: 'Placeholder image',
          },
        ],
      },
      {
        id: 3,
        name: 'Gallery 3',
        images: [
          {
            id: 1,
            src: _16x9,
            alt: 'Placeholder image',
          },
          {
            id: 2,
            src: _16x9,
            alt: 'Placeholder image',
          },
          {
            id: 3,
            src: _4x3,
            alt: 'Placeholder image',
          },
          {
            id: 4,
            src: _1x1,
            alt: 'Placeholder image',
          },
        ],
      },
    ],
  },
};
