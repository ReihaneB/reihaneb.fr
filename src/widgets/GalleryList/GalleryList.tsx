import Separator from '@/components/Separator/Separator';

import Gallery from '../Gallery/Gallery';

import type { GalleryListProps } from './GalleryList.d';
import styles from './GalleryList.module.css';

function GalleryList({
  galleries,
}: GalleryListProps) {
  return (
    <>
      <div className={styles.separatorContainer}>
        <Separator text="Projets majeurs" />
      </div>
      <ul className={styles.galleryList}>
        {galleries.map(({ id, name, description, images }) => (
          <li key={id}>
            <Gallery
              name={name}
              images={images}
              description={description}
            />
          </li>
        ))}
      </ul>
    </>
  );
}

export default GalleryList;
