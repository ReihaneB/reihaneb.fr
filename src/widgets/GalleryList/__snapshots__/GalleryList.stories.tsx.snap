// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widgets/GalleryList _1Gallery smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-widgets-GalleryList-GalleryList-module__separatorContainer--Qq3vm">
    <hr class="src-components-Separator-Separator-module__root--N5eOl"
        data-testid="separator-component"
        style="--separator-text: &quot;Projets majeurs&quot;; --separator-text-length: 15ch;"
    >
  </div>
  <ul class="src-widgets-GalleryList-GalleryList-module__galleryList--UNK7W">
    <li>
      <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
        Gallery 1
      </h2>
      <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1920"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
               src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1920"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
               src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
      </ul>
    </li>
  </ul>
</main>
`;

exports[`Widgets/GalleryList _2Galleries smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-widgets-GalleryList-GalleryList-module__separatorContainer--Qq3vm">
    <hr class="src-components-Separator-Separator-module__root--N5eOl"
        data-testid="separator-component"
        style="--separator-text: &quot;Projets majeurs&quot;; --separator-text-length: 15ch;"
    >
  </div>
  <ul class="src-widgets-GalleryList-GalleryList-module__galleryList--UNK7W">
    <li>
      <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
        Gallery 1
      </h2>
      <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1920"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
               src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1920"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
               src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
      </ul>
    </li>
    <li>
      <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
        Gallery 2
      </h2>
      <p class="src-widgets-Gallery-Gallery-module__description--cwB1k">
        Description
      </p>
      <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1920"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
               src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1920"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
               src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="800"
               height="600"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_4x3.58bee57f.png?w=828&amp;q=75 1x, /static/media/_4x3.58bee57f.png?w=1920&amp;q=75 2x"
               src="/static/media/_4x3.58bee57f.png?w=1920&amp;q=75"
               style="color: transparent;"
          >
        </li>
      </ul>
    </li>
  </ul>
</main>
`;

exports[`Widgets/GalleryList _3Galleries smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-widgets-GalleryList-GalleryList-module__separatorContainer--Qq3vm">
    <hr class="src-components-Separator-Separator-module__root--N5eOl"
        data-testid="separator-component"
        style="--separator-text: &quot;Projets majeurs&quot;; --separator-text-length: 15ch;"
    >
  </div>
  <ul class="src-widgets-GalleryList-GalleryList-module__galleryList--UNK7W">
    <li>
      <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
        Gallery 1
      </h2>
      <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1920"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
               src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1920"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
               src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
      </ul>
    </li>
    <li>
      <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
        Gallery 2
      </h2>
      <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1920"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
               src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1920"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
               src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="800"
               height="600"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_4x3.58bee57f.png?w=828&amp;q=75 1x, /static/media/_4x3.58bee57f.png?w=1920&amp;q=75 2x"
               src="/static/media/_4x3.58bee57f.png?w=1920&amp;q=75"
               style="color: transparent;"
          >
        </li>
      </ul>
    </li>
    <li>
      <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
        Gallery 3
      </h2>
      <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1920"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
               src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1920"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
               src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="800"
               height="600"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_4x3.58bee57f.png?w=828&amp;q=75 1x, /static/media/_4x3.58bee57f.png?w=1920&amp;q=75 2x"
               src="/static/media/_4x3.58bee57f.png?w=1920&amp;q=75"
               style="color: transparent;"
          >
        </li>
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1080"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_1x1.ea48a62d.png?w=1080&amp;q=75 1x, /static/media/_1x1.ea48a62d.png?w=3840&amp;q=75 2x"
               src="/static/media/_1x1.ea48a62d.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
      </ul>
    </li>
  </ul>
</main>
`;

exports[`Widgets/GalleryList Playground smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-widgets-GalleryList-GalleryList-module__separatorContainer--Qq3vm">
    <hr class="src-components-Separator-Separator-module__root--N5eOl"
        data-testid="separator-component"
        style="--separator-text: &quot;Projets majeurs&quot;; --separator-text-length: 15ch;"
    >
  </div>
  <ul class="src-widgets-GalleryList-GalleryList-module__galleryList--UNK7W">
    <li>
      <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
        Gallery 1
      </h2>
      <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1080"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_1x1.ea48a62d.png?w=1080&amp;q=75 1x, /static/media/_1x1.ea48a62d.png?w=3840&amp;q=75 2x"
               src="/static/media/_1x1.ea48a62d.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1920"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
               src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
      </ul>
    </li>
    <li>
      <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
        Gallery 2
      </h2>
      <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1920"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
               src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="1920"
               height="1080"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
               src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
               style="color: transparent;"
          >
        </li>
        <li>
          <img alt="Placeholder image"
               fetchpriority="high"
               width="800"
               height="600"
               decoding="async"
               data-nimg="1"
               class="src-components-Image-Image-module__image--gXjYq"
               srcset="/static/media/_4x3.58bee57f.png?w=828&amp;q=75 1x, /static/media/_4x3.58bee57f.png?w=1920&amp;q=75 2x"
               src="/static/media/_4x3.58bee57f.png?w=1920&amp;q=75"
               style="color: transparent;"
          >
        </li>
      </ul>
    </li>
  </ul>
</main>
`;
